kaolin/transforms/transforms.py

```diff
@@ -36,7 +36,45 @@
 from kaolin.transforms import voxelfunc
 
 
-class Compose(object):
+class Transform(object):
+    """Base class for all Kaolin transforms.
+
+    This class generates __repr__ string automatically. Given that all attributes
+    have valid __repr__, it generates a string with the following format:
+    .. code-block::
+       MyClass(a=1, b=2, ...)
+
+    The attributes are sorted alphabetically to ensure determinism.
+
+    To exclude certain attributes from appearing in the __repr__, define a list
+    "__ignored_params__" in the subclass (not the instance). Example:
+    .. code-block::
+       class MyClass(Transform):
+           __ignored_params__ = ['a', 'b']
+
+    NOTE:
+    - Since the __repr__ generation depends on the __repr__ of attributes,
+      objects that have incorrect (such as arbitrary objects that output
+      their memory address by default) or indeterministic (such as
+      dictionaries) __repr__ should not be assigned as attribute.
+      If unavoidable, override __repr__ manually.
+    """
+
+    def __repr__(self):
+        ignored = set(getattr(self.__class__, '__ignored_params__', []))
```

Comment:
Why not just `ignored = set(getattr(self, '__ignored_params__', []))` ? It would works for both class and instance attribute (if for instance there is some conditional params to ignore or something like that).

Reply:
That's a good point. My concern is that if the object itself can have `__ignored_params__` then you would also have to put `__ignored_params__` either in the class's `__ignored_params__` or the object's `__ignored_params__`, which might be a little error-prone. Maybe we should avoid this entirely by just using the class attribute?

Reply:
Well it's python, stubborn users will always have possibility to stupid things :D
I would assume that the user who would create his own class would know what he's doing.

The base class is not enforcing class or instantiate attribute neither, so it's not misleading to let the user do whatever he wants

Reply:
> > > Should we really ignore the params `eps` and `inplace` ? What if `inplace` is incorrectly used (like it actually modify a tensor that was supposed to be reused), what if `eps` have a strong impact on the output ?
> > > I'm actually struggling to see cases where you can ignore an attribute, if you change an attribute to a transform I assume you actually want to change the output right ? If your attribute for instance is only speed or memory related, I would argue that the most efficient way to have the output is.... to use the cache (so don't change the attributes) !
> > > Do you have an example ?
> > 
> > 
> > That's actually a good point. I will remove `inplace` and `eps` from the list.
> 
> But then is there any use at `__ignore_params__` ?

Maybe here?
https://github.com/NVIDIAGameWorks/kaolin/blob/8937233ba30059eb3d003d1add0fad1ee54c4c1a/kaolin/transforms/transforms.py#L108

Reply:
Good point. Fixed.